pkgs/development/compilers/cudatoolkit/common.nix

```diff
@@ -178,7 +178,7 @@ backendStdenv.mkDerivation rec {
   ];
 
   preFixup = ''
-    patchelf $out/lib64/libnvrtc.so --add-needed libnvrtc-builtins.so
+    patchelf $(find "$out" -type l -name libnvrtc.so -execdir readlink -f {} \; | sort -u) --add-needed libnvrtc-builtins.so
```

Comment:
I can do that, although we'd need to check for the highest version first.

Reply:
oh yeah good point. `versions.major version == "10"` is prob the way to go... at least I'm guessing that the location is consistent across major versions

Reply:
In case you like it better, I'm leaving this option here. However, I plan to do the if-based one.
```bash
# pick the file that exists; if none exist patchelf will fail with missing argument
patchelf $(ls "$out/lib/a" "$out/lib/b" 2>/dev/null) --add-needed libnvrtc.so
```

Reply:
Going with `lib.versionOlder version "11"` as that is used elsewhere in the file.

Reply:
Also consider using

```
while IFS= read ...; do
  ...
done < <(find ...)
```

instead of `$(find ...)`. This avoids spawning a sub-shell and should handle whitespaces correctly (although we don't expect any)

Reply:
The original line you're commenting on is no longer in this PR.